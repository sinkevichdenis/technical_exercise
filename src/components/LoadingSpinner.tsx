import React from 'react';
import {
  ActivityIndicator,
  StyleProp, View, ViewStyle
} from 'react-native';

import { Colors } from 'styles/colors';


type Props = {
  containerStyle?: StyleProp<ViewStyle>,
}

export const LoadingSpinner: React.FC<Props> = React.memo(({
  containerStyle,
}) => (
  <View
    style={ containerStyle }
    testID='spinner'
  >
    <ActivityIndicator
      size="large"
      color={ Colors.BLUE }
    />
  </View>
));
