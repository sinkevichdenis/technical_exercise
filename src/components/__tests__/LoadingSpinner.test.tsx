import { render } from '@testing-library/react-native';
import React from 'react';

import { LoadingSpinner } from '../LoadingSpinner';


describe('LoadingSpinner component', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<LoadingSpinner />);
    expect(toJSON()).toMatchSnapshot();
  });
});
