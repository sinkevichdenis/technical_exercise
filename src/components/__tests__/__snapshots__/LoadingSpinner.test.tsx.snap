// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component should match snapshot 1`] = `
<View>
  <ActivityIndicator
    color="#007ff9"
    size="large"
  />
</View>
`;
